lively.halos/morph.js

```diff
@@ -1375,12 +1375,17 @@ class CopyHaloItem extends HaloItem {
     const { halo } = this; const { target } = halo; const world = halo.world();
     const isMultiSelection = target instanceof MultiSelectionTarget;
     halo.remove();
-
     connect(hand, 'update', this, 'update');
-
     if (isMultiSelection) {
       // FIXME! haaaaack
-      const copies = target.selectedMorphs.map(ea => world.addMorph(ea.copy()));
+      const copies = target.selectedMorphs.map(ea => {
+        if (ea.canBeCopied()) {
+          const copy = ea.copy(true);
+          world.addMorph(copy);
+          return copy;
+        }
+      }).filter(copy => copy);
```

Comment:
what is happening here?

Reply:
We chain `map` and `filter` afterwards, to only keep the entries of the Array that `map` produced that we are actually interested in. 
The code in `map` gets executed for every entry in `target.selectedMorphs` and produces an item in the resulting array for each entry aswell. These are either actual copies that we want to keep or undefined (or null or something, in any case falsy, do not quote me on the specifics). This is exactly what we use in the `filter` call, where the condition for an item to stay in the array basically is "copy should be truthy", which is the case for actual copies but not for the entries that do not enter the `if` block.

Reply:
thanks a lot!